src/cascadia/WpfTerminalControl/TerminalContainer.cs

```diff
@@ -0,0 +1,247 @@
+                        }
+                        else
+                        {
+                            NativeMethods.SendTerminalOutput(this.terminal, Clipboard.GetText());
+                        }
+                        break;
+                    case PInvoke.User32.WindowMessage.WM_MOUSEMOVE:
+                        this.MouseMoveHandler((int)wParam, (int)lParam);
+                        break;
+                    case PInvoke.User32.WindowMessage.WM_KEYDOWN:
+                        NativeMethods.ClearSelection(this.terminal);
+                        break;
+                    case PInvoke.User32.WindowMessage.WM_CHAR:
+                        this.HandleChar((char)wParam);
+                        break;
+                    case PInvoke.User32.WindowMessage.WM_WINDOWPOSCHANGED:
+                        var windowpos = (NativeMethods.WINDOWPOS)Marshal.PtrToStructure(lParam, typeof(NativeMethods.WINDOWPOS));
+                        if (((PInvoke.User32.SetWindowPosFlags)windowpos.flags).HasFlag(PInvoke.User32.SetWindowPosFlags.SWP_NOSIZE))
+                        {
+                            break;
+                        }
+
+                        NativeMethods.TriggerResize(this.terminal, windowpos.cx, windowpos.cy, out int columns, out int rows);
+                        this.connection?.Resize((uint)rows, (uint)columns);
+                        break;
+                    case PInvoke.User32.WindowMessage.WM_MOUSEWHEEL:
+                        var delta = (((int)wParam) >> 16);
+                        this.UserScrolled?.Invoke(this, delta);
+                        break;
+                }
+            }
+
+            return IntPtr.Zero;
+        }
+
+        private void LeftClickHandler(int lParam)
+        {
+            var altPressed = NativeMethods.GetKeyState((int)PInvoke.User32.VirtualKey.VK_MENU) < 0;
+            var x = (short)(((int)lParam << 16) >> 16);
+            var y = (short)((int)lParam >> 16);
+            PInvoke.COORD cursorPosition = new PInvoke.COORD()
+            {
+                X = x,
+                Y = y,
+            };
+
+            NativeMethods.StartSelection(this.terminal, cursorPosition, altPressed);
+        }
+
+        private void MouseMoveHandler(int wParam, int lParam)
+        {
+            if (((int)wParam & 0x0001) == 1)
+            {
+                var x = (short)(((int)lParam << 16) >> 16);
+                var y = (short)((int)lParam >> 16);
+                PInvoke.COORD cursorPosition = new PInvoke.COORD()
+                {
+                    X = x,
+                    Y = y,
+                };
+                NativeMethods.MoveSelection(this.terminal, cursorPosition);
+            }
+        }
+
+        private bool HandleChar(char character)
+        {
+            if (this.connection == null)
+            {
+                return false;
+            }
+
+            if (char.IsHighSurrogate(character))

```

Comment:
We have an intense amount of code already elsewhere in the codebase related to what to do with invalid unicode sequences and how to correct them. Why is there another one here?